controllers/managedfusionoffering_controller.go

```diff
@@ -65,11 +72,42 @@ func (r *ManagedFusionOfferingReconciler) SetupWithManager(mgr ctrl.Manager, ctr
 		Owns(&opv1a1.CatalogSource{}).
 		Owns(&opv1a1.Subscription{})
 
-	pluginSetupWatches(controllerBuilder)
+	if doesCRDsExists() {
+		r.DoesCRDExistsAtStart = true
+		pluginSetupWatches(controllerBuilder)
+	}
```

Comment:
We have an established way to check for CRD existence. It was added with the SDN/OVN contribution a couple of months back on the original deployer repo. Please use this

https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/controllers/managedocs_controller.go#L323-L328


Also, why is this implemented at the offering reconciler level where it is a fix for df add watches? This should go into dfSetupWatches. This all hack should be internal to the DF offering.